core/java/iesi-core/src/main/java/io/metadew/iesi/script/action/wfa/WfaExecuteQueryPing.java

```diff
@@ -116,11 +116,21 @@ protected boolean executeAction() throws InterruptedException {
         boolean setRuntimeVariables = converSetRuntimeVariable(getSetRuntimeVariables().getValue());
         int timeoutInterval = convertTimeoutInterval(getTimeoutInterval().getValue());
         int waitInterval = convertWaitInterval(getWaitInterval().getValue());
-        Connection connection = ConnectionConfiguration.getInstance()
-                .get(new ConnectionKey(connectionName, this.getExecutionControl().getEnvName()))
-                .get();
+
+        Connection connection;
+        try {
+            connection = ConnectionConfiguration.getInstance()
+                    .get(new ConnectionKey(connectionName, this.getExecutionControl().getEnvName()))
+                    .get();
+        } catch (Exception e) {
+            throw new RuntimeException("Unknown connection name");
+        }
+
         ConnectionOperation connectionOperation = new ConnectionOperation();
         Database database = DatabaseHandler.getInstance().getDatabase(connection);
+        if (database == null) {
+            throw new RuntimeException("Error establishing DB connection");
+        }
```

Comment:
DatabaseHandler.getInstance().getDatabase(...) will never return a null object. It will either return a Database object or throw an Exception. There should be no need to check parameter for null

Reply:
removed